dist/scripts/main.js

```diff
@@ -0,0 +1,157 @@
+      someInput.classList.remove('inputs-column_valid');
+      someInput.classList.add('inputs-column_invalid');
+    };
+    const backgroundGreen = function(someInput) {
+      someInput.classList.remove('inputs-column_invalid');
+      someInput.classList.add('inputs-column_valid');
+    };
+    const emptied = function(someInput) {
+      someInput.classList
+        .remove('inputs-column_invalid', 'inputs-column_invalid');
+    };
+
+    isValid = fullNameRegExp.test(fullName.value.trim());
+    if (fullName.value !== '') {
+      if (isValid) {
+        backgroundGreen(fullName);
+      } else {
+        backgroundRed(fullName);
+      }
+    } else {
+      emptied(fullName);
+    }
+    if (!needMore.checked) {
+      const nextResult = phoneNumberRegExp.test(phoneNumber.value.trim());
+      if (phoneNumber.value !== '') {
+        if (nextResult) {
+          backgroundGreen(phoneNumber);
+        } else {
+          backgroundRed(phoneNumber);
+        }
+      } else {
+        emptied(phoneNumber);
+      }
+      isValid &= nextResult;
+      isValid &= region.options.selectedIndex !== 0;
+      isValid &= city.options.selectedIndex !== 0;
+    }
+    return isValid;
+  };
+
+  const renderByCheckBox = function() {
+    phoneNumber.hidden = needMore.checked;
+    document.querySelector(`label[for='${phoneNumber.id}']`)
+      .hidden = needMore.checked;
+    region.hidden = needMore.checked;
+    document.querySelector(`label[for='${region.id}']`)
+      .hidden = needMore.checked;
+    city.hidden = needMore.checked;
+    document.querySelector(`label[for='${city.id}']`)
+      .hidden = needMore.checked;
+  };
+
+  // Initializing region and city selects
+  initBlankInSelect(region);
+  initBlankInSelect(city);
+  Object.keys(citiesByRegionsObject).forEach((someRegion) => {
+    region.options[region.options.length] = new Option(someRegion, someRegion);
+  });
+  region.addEventListener('change', (someEvent) => {
+    const selectedItem = someEvent.currentTarget
+      .options[region.options.selectedIndex];
+    if (selectedItem.value === 'blank') { return; }
+
+    citiesByRegionsObject[selectedItem.value].forEach((someCity) => {
+      city.options[city.options.length] = new Option(someCity, someCity);
+    });
+    city.options[city.options.length] = new Option(
+      'Not in the list', 'Other city'
+    );
+  });
+
+  const validateAndRender = function(someEvent) {
+    if (someEvent) {
+      someEvent.target.classList.remove('inputs-column_inputting');
+    }
+    if (validateAll()) {
+      submitButton.disabled = false;
+      submitButton.classList.add('hover');
+    } else {
+      submitButton.disabled = true;
+      submitButton.classList.remove('hover');
+    }
+  };
+
+  // Add trigger for listing changing checkbox and rendering the form
+  // according to it's value
+  needMore.addEventListener('change', renderByCheckBox);
+
+  registrationForm.addEventListener('keyup', (someEvent) => {
```

Comment:
For my mind you do not need it, you replace native functionality that form have

Reply:
I don't know why but ufter pressing Enter there was no focus changing. And on check there was not checking... That why I wrote this listner.